proxy/hdrs/URL.cc

```diff
@@ -1846,4 +1823,42 @@ REGRESSION_TEST(VALIDATE_HDR_FIELD)(RegressionTest *t, int /* level ATS_UNUSED *
   }
 }
 
+
+const static struct {
+  const char *const uri;
+  bool valid;
+} http_strict_uri_parsing_test_case[] = {{"/home", true},
+                                         {"/path/data?key=value#id", true},
+                                         {"/ABCDEFGHIJKLMNOPQRSTUVWXYZ", true},
+                                         {"/abcdefghijklmnopqrstuvwxyz", true},
+                                         {"/0123456789", true},
+                                         {":/?#[]@", true},
+                                         {"!$&'()*+,;=", true},
+                                         {"-._~", true},
+                                         {"%", true},
+                                         {"\n", false},
+                                         {"\"", false},
+                                         {"<", false},
+                                         {">", false},
+                                         {"\\", false},
+                                         {"^", false},
+                                         {"`", false},
+                                         {"{", false},
+                                         {"|", false},
+                                         {"}", false},
+                                         {"é", false}};
+
+REGRESSION_TEST(STRICT_URI_PARSING)(RegressionTest *t, int /* level ATS_UNUSED */, int *pstatus)
```

Comment:
We don't use all caps ``STRICT_URI_PARSING``.  How about ``ParseRules_strict_URI`` or something?